### PR TITLE
feat: #tno-1522 - refactor Product into Media Type

### PR DESCRIPTION
**Pull request by kylgarmor**
- updated all backend code
- created db migration for postgres
- created migration for elastic
- created new npm library version
- updated all subscriber code
- updated all editor code

### CODE REVIEW

File: app/subscriber/src/store/hooks/lookup/useLookup.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ import {
   ITopicModel,
   ITopicScoreRuleModel,
   IUserModel,
+  StorageKeys,
```
Comment (Fosol) [inline app/subscriber/src/store/hooks/lookup/useLookup.ts:24]: Nice use of a constant

File: libs/net/dal/Migrations/20231108235551_1.0.89.cs [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+{
+    /// <inheritdoc />
+    public partial class _1089 : SeedMigration
+    {
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            PreUp(migrationBuilder);
+
+            migrationBuilder.DropForeignKey(
```
Comment (Fosol) [inline libs/net/dal/Migrations/20231108235551_1.0.89.cs:19]: Looks like it correctly renamed :D

File: libs/net/models/Areas/Admin/DataLocation/ConnectionModel.cs [hunk 3]
```diff
@@ -5,7 +5,7 @@
 namespace TNO.API.Areas.Admin.Models.DataLocation;
 
 /// <summary>
-/// ConnectionModel class, provides a model that represents an product.
+/// ConnectionModel class, provides a model that represents an connection.
```
Comment (Fosol) [inline libs/net/models/Areas/Admin/DataLocation/ConnectionModel.cs:8]: A extra benefit of this refactor, fixing comments :D

File: libs/net/models/Areas/Subscriber/Models/Content/SourceModel.cs [hunk 4]
```diff
@@ -34,7 +34,7 @@ public class SourceModel : BaseTypeModel<int>
     /// <summary>
     /// get/set -
     /// </summary>
-    public int? ProductId { get; set; }
+    public int? MediaTypeId { get; set; } //fixed
```
Comment (Fosol) [inline libs/net/models/Areas/Subscriber/Models/Content/SourceModel.cs:37]: What's the comment for?

File: libs/net/models/Areas/Subscriber/Models/Contributor/SourceModel.cs [hunk 5]
```diff
@@ -49,7 +49,7 @@ public class SourceModel
     /// <summary>
     /// get/set -
     /// </summary>
-    public int? ProductId { get; set; }
+    public int? MediaTypeId { get; set; } //fixed
```
Comment (Fosol) [inline libs/net/models/Areas/Subscriber/Models/Contributor/SourceModel.cs:52]: What's the comment for?

File: libs/npm/core/src/hooks/api-editor/constants/Settings.ts [hunk 6]
```diff
@@ -8,6 +8,6 @@ export const Settings = {
   TopStoryAlert: 'TopStoryAlertId',
   MorningReport: 'MorningReportId',
   FrontPageImagesReport: 'FrontPageImagesReportId',
-  NewsRadioProductFilter: 'NewsRadioProductFilterId',
+  NewsRadioMediaTypeFilter: 'NewsRadioMediaTypeFilterId',
```
Comment (Fosol) [inline libs/npm/core/src/hooks/api-editor/constants/Settings.ts:11]: Make sure you manually update DEV and TEST for this

**Review by Fosol — COMMENTED** [review]
You'll need to publish `tno-core` and fix the other build issues.
Deploying may be difficult.  If the Indexing Service has Kafka Lag it may attempt to index old content data.  You also don't want it to start indexing new data sent from the API until the migration is completed.
We don't have the Elasticsearch migration in the main pipeline atm.  You have to manually run the elastic migration pipeline separately.  Might be worth it to add it to the main pipeline so that it's all done at one time.

File: libs/npm/core/src/hooks/api-editor/constants/Settings.ts (hunk 6 above)
Comment (kylgarmor) [reply libs/npm/core/src/hooks/api-editor/constants/Settings.ts:11]: I'll ask you how when you merge the PR ;)

File: libs/net/models/Areas/Admin/DataLocation/ConnectionModel.cs (hunk 3 above)
Comment (kylgarmor) [reply libs/net/models/Areas/Admin/DataLocation/ConnectionModel.cs:8]: my pleasure

File: app/subscriber/src/store/hooks/lookup/useLookup.ts (hunk 1 above)
Comment (kylgarmor) [reply app/subscriber/src/store/hooks/lookup/useLookup.ts:24]: I know right? ;)

File: app/editor/src/features/content/papers/PaperToolbar.tsx [hunk 7]
```diff
@@ -36,13 +36,13 @@ export interface IPaperToolbarProps {
 export const PaperToolbar: React.FC<IPaperToolbarProps> = ({ onSearch }) => {
   const navigate = useNavigate();
   const [{ filterPaper: filter }, { storeFilterPaper }] = useContent();
-  const [{ productOptions: pOptions }] = useLookupOptions();
+  const [{ mediaTypeOptions: mediaTypeOptions }] = useLookupOptions();
```
Comment (Fosol) [inline app/editor/src/features/content/papers/PaperToolbar.tsx:39]: Fix this

**Review by Fosol — APPROVED** [review]
Nice work

**Pull request merged**